ui/src/app/pages/tanagra-dev/tanagra-dev.tsx

```diff
@@ -1,30 +1,85 @@
 import * as React from 'react';
 
+import { Button } from 'app/components/buttons';
+import { SpinnerOverlay } from 'app/components/spinners';
 import { withSpinnerOverlay } from 'app/components/with-spinner-overlay';
+import { serverConfigStore } from 'app/utils/stores';
 
-const { useEffect } = React;
+const { useCallback, useEffect, useState } = React;
+
+
+export function useExitActionListener(callback: () => void) {
+  const listener = useCallback(
+    (event: MessageEvent) => {
+      const tanagraUrl = serverConfigStore.get().config.tanagraBaseUrl;
+      if (
+        event.origin !== tanagraUrl ||
+        typeof event.data !== 'object' ||
+        event.data.message !== 'CLOSE'
+      ) {
+        return;
+      }
+      callback();
+    },
+    [callback]
+  );
+
+  useEffect(() => {
+    window.addEventListener('message', listener);
+    return () => {
+      window.removeEventListener('message', listener);
+    };
+  }, [listener]);
+}
 
 export const TanagraDev = withSpinnerOverlay()(({ hideSpinner }) => {
+  const [loadingIframe, setLoadingIframe] = useState(false);
+  const [showIframe, setShowIframe] = useState(false);
   useEffect(() => {
     hideSpinner();
   }, []);
 
-  return (
+  useExitActionListener(() => {
+    setShowIframe(false);
+  });
+
+  // Temporary variable to hardcode existing study ids until we have a link between workspaces and studies
+  const studyId =
+    process.env.REACT_APP_ENVIRONMENT === 'local' ? 'tqmXfZ4qzu' : 'UslvvbIYxk';
```

Comment:
I thought we will preload the study table with workspaceId for the Id column, then we would not have to save study for the demo.?

Reply:
Changed to use the workspace namespace, which will also be the Tanagra study id.